lib/processors/jsdoc/lib/ui5/plugin.js

```diff
@@ -516,6 +523,192 @@ function isPotentialEnum(node) {
 	return node.properties.every((prop) => isCompileTimeConstant(prop.value));
 }
 
+// ---- ES6+ Destructuring ---------------------------------------------------------
+/**
+ * Checks against currentModule.localNames to verify an enum imported from an external module
+ * 
+ * @param {Array} varsDestructuringStack 
+ * @returns {Object}
+ */
+function checkEnumsFromExternalModules(varsDestructuringStack) {
+	const localName = varsDestructuringStack[0];
+	const potentialEnumKey =
+		varsDestructuringStack[varsDestructuringStack.length - 1];
+
+	const curModuleNames = currentModule.localNames[localName] || {};
+	const localModule = curModuleNames.module || curModuleNames.class;
+
+	// Find if there are enums defined for that module
+	return Object.keys(enumValues)
+		.filter((curEnumKey) => {
+			const [enums, resource] = curEnumKey.split("||");
+
+			// Check the resource
+			if (resource.indexOf(localModule) === -1) {
+				return false;
+			}
+
+			// Check if the key is in the enums
+			return enums.split("|").includes(potentialEnumKey);
```

Comment:
Not sure whether this is the correct way to check this as this filtering relies on ENUM's:
1. Filename of where it's been defined (this one is ok)
2. Check whether any of the enums in the file contains a such element.

It should be fine as the chain it creates is:
{dependencyName}.{enumName}.{enumElement} i.e. library.XXX.Foo
Foo could be defined in other enums places in the same file, but the check is just whether it's there or not. The XXX (eunm's name) is derived in the previous step by traversing up to the tree in the previous step.